### DISCUSSION TITLE
Delete refresh tokens from database

### DISCUSSION THREAD

**Discussion by manusabarros** [supabase/supabase#2937] [Questions] [answered]
For how long refresh tokens are stored in `auth.refresh_tokens` table? Or do they stay indefinitely?(and if so, is it safe to delete the old ones?)

**Comment by kangmingtay** [comment, marked as answer]
Hey @manusabarros, the refresh tokens are stored in the `auth.refresh_tokens` table forever unless you remove them. There is a field called `revoked` which determines whether a refresh token has been revoked or not. Also, the refresh token is only valid for a one-time use and then it will be revoked. 

Hope this helps!
